ballle98/AqualinkD#13: add time sync

ballle98 opened the pull request:
fix #93

ballle98 commented on pda_aq_programmer.c:
```diff
@@ -858,14 +852,46 @@ bool set_PDA_aqualink_freezeprotect_setpoint(struct aqualinkdata *aq_data, int v
 }
 
+bool set_PDA_aqualink_time(struct aqualinkdata *aq_data) {
+  if (! goto_pda_menu(aq_data, PM_SET_TIME)) {
```
going to SET_TIME menu will update aq_data->tm. And every time the menu is updated the tm value is also updated

sfeakes commented:
I'll go over the changes what I get some time.  But previous implementation was inefficient, that's why I asked for changes and for it to fit within the current code structure.
Remember you get time as a string, pass it to the UI as a string, and update it as a string, and there is no reason to check it every time you get a new time.  So using the currently implemented logic, the time string only get's converted to time datatype every hour or so when you need to check the time is in-sync (might be every 10 mins, I can't remember).  But at that check is the only time you actually need to convert the time string.

ballle98 commented on pda_aq_programmer.c:
```diff
@@ -966,132 +962,41 @@ Debug:   PDA:       PDA Menu Line 8 = Press SELECT
-    send_cmd(KEY_PDA_BACK);
-    send_cmd(KEY_PDA_BACK);
-    return true;
+  if (strptime(pda_m_line(2), "%t%D %a", &panel_tm) == NULL) {
```
removed tm from aq_data now getting panel time from menu buffer

ballle98 commented:
@sfeakes I have rebased and updated this pull request to the latest version.

I have tested this thoroghly with my PDA pannel and it synce the time correctly.

I have granted permision to for you to modify this pull requests if you want to change anything before merging
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

There are instructions for working with the pull offline on a branch again you can make changes in additional commits before merging
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally?tool=webui

attached logs showing it working

[stdout-20230528_2203.txt](https://github.com/sfeakes/AqualinkD/files/11587905/stdout-20230528_2203.txt)

ballle98 commented on pda.c:
```diff
@@ -307,18 +307,20 @@ void process_pda_packet_msg_long_time(const char *msg)
   if (msg[AQ_MSGLEN - 1] == ' ')
   {
-    strncpy(_aqualink_data->time, msg + 9, 6);
+    snprintf(_aqualink_data->time, sizeof(_aqualink_data->time), "%.6s", msg + 9);
```
strncpy is an unsafe string function.  it assumes the src is null terminated
